### ISSUE TITLE
Content is corrupted in case of utf8 multibyte characters

### ISSUE THREAD

**Issue by Biktop** [mholt/PapaParse#743]
I found that in version 5.1.0 downloaded cvs parsed incorrectly in case if source data contains utf8 multibyte characters (whereas 5.0.2 didn't have this issue). I noticed that PapaParse tried to download second chunk even if there no any extra bytes:

First chunk:
```
< Content-Length: 184027
< Content-Range: bytes 0-184026/184027
```

Second chunk (not necessary)

```
> Range: bytes=183546-5426425
< Content-Length: 481
< Content-Range: bytes 183546-184026/184027
```
After some digging I think that problem [here](https://github.com/mholt/PapaParse/blob/5219809f1d83ffa611ebe7ed13e8224bcbcf3bd7/papaparse.js#L685)
`this._start += xhr.responseText.length;`

Because _Content-Length_ can be greater then text length (because some characters can be 2 or more bytes).

 I think it should be something like this:
`this._start += +xhr.getResponseHeader('Content-Length');`

**Comment by pokoli** [comment]
I think this should be fixed by #745 could you please test it?

**Comment by Biktop** [comment]
I've tested it and it seems that it works!

**Comment by pokoli** [comment]
Ok, so I've merged #745 which should fix this

**Comment by Biktop** [comment]
Are going to publish this fix soon?

**Comment by pokoli** [comment]
I've just published 5.1.1 which contains this fix!
